distro/runtime/concierge/start.sh

```diff
@@ -1,51 +1,198 @@
+                 # by default, start recording for 5 min, put to userdata folder
+                 JFR_OPTS="$JFR_OPTS -XX:StartFlightRecording=duration=300s,filename=$BASE_FOLDER/userdata/smarthome-concierge.jfr"
+            fi
+       fi
+    fi
+}
+
+# arg 0: debug enabled: true/false
+setJmxOptions() {
+    # enable in debug mode JMX when running on compact3/fulljre
+    JMX_OPTS=""    
+    if [ "$1" = "true" ] ; then
+        if [ "$JAVA_COMPACT_PROFILE" = "compact3" -o "$JAVA_COMPACT_PROFILE" = "fulljre" ]; then
+            # compact3, fulljre: enable JMX in debug mode, needed to flight recorder too
+            JMX_OPTS="$JMX_OPTS -Dcom.sun.management.jmxremote.port=7091"
+            JMX_OPTS="$JMX_OPTS -Dcom.sun.management.jmxremote.rmi.port=7091"
+            JMX_OPTS="$JMX_OPTS -Dcom.sun.management.jmxremote.authenticate=false"
+            JMX_OPTS="$JMX_OPTS -Dcom.sun.management.jmxremote.ssl=false"
+        fi
+    fi
+}
+
+# arg 0: debug enabled: true/false
+setDebugOptions() {
+    JAVA_DEBUG_OPTS=""
+    if [ "$1" = "true" ] ; then
+        if [ "$JAVA_VENDOR" = "Azul" -a "$JAVA_COMPACT_PROFILE" = "compact2" ] ; then
+            # will NOT be supported, $JAVA_HOME/lib/libjdwp.so is missing
+            :
+        else
+            DEFAULT_JAVA_DEBUG_PORT="5005"
+            if [ "x$JAVA_DEBUG_PORT" = "x" ]; then
+                JAVA_DEBUG_PORT="$DEFAULT_JAVA_DEBUG_PORT"
+            fi
+            DEFAULT_JAVA_DEBUG_OPTS="-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$JAVA_DEBUG_PORT"
+            JAVA_DEBUG_OPTS="$JAVA_DEBUG_OPTS $DEFAULT_JAVA_DEBUG_OPTS"
+        fi
+
+        # chatty logging setttings
+        JAVA_DEBUG_OPTS="$JAVA_DEBUG_OPTS -Dlogback.configurationFile=$RUNTIME_FOLDER/etc/logback_debug.xml"
+    else
+        # more silent logging setttings
+        JAVA_DEBUG_OPTS="$JAVA_DEBUG_OPTS -Dlogback.configurationFile=$RUNTIME_FOLDER/etc/logback.xml"
+    fi
+}
+
+run() {
+    DEBUG_ENABLED=false
+    while [ "$1" != "" ]; do
+        case $1 in
+            'debug')
+                DEBUG_ENABLED=true
+                shift
+                ;;
+            *)
+                break
+                ;;
+        esac
+    done
+
+    findJvmVersion $DEBUG_ENABLED
+    checkJvmVersion
+
+    setDebugOptions $DEBUG_ENABLED
+    setFlightRecorderOptions $DEBUG_ENABLED
+    setJmxOptions $DEBUG_ENABLED
+
+    # Find the concierge framework jar
+    MAIN=$(find framework -name "org.eclipse.concierge-5.0.0*.jar" | sort | tail -1);
+
+    # show java command when debug enabled 
+    if [ "$DEBUG_ENABLED" = "true" ] ; then
+        # echo "JAVA_OPTS=$JAVA_OPTS"
+        # echo "JAVA_DEBUG_OPTS=$JAVA_DEBUG_OPTS"
+        # echo "JFR_OPTS=$JFR_OPTS"
+        # echo "JMX_OPTS=$JMX_OPTS"
+        set -x
+    fi
+
+    $JAVA_BIN $JAVA_OPTS $JAVA_DEBUG_OPTS $JFR_OPTS $JMX_OPTS \
+	    -Djava.awt.headless=true \
+	    -Djava.library.path=lib \
+	    -Dfile.encoding="UTF-8" \
+	    -Dosgi.clean=true \
```

Comment:
You didn't introduce this, but I see it here: Do we really want this option? Afaik, this will wipe all settings in config admin that might have been done through the Paper UI - probably not a desired behavior?